inject/src/main/java/io/micronaut/inject/processing/BeanDefinitionBuilderFactory.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.inject.processing;
+
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.DefaultScope;
+import io.micronaut.context.annotation.Executable;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Property;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.core.annotation.AnnotationMetadata;
+import io.micronaut.core.annotation.AnnotationUtil;
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.inject.ProcessingException;
+import io.micronaut.inject.ast.ClassElement;
+import io.micronaut.inject.ast.ElementQuery;
+import io.micronaut.inject.visitor.VisitorContext;
+
+import java.lang.annotation.Annotation;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Bean definition builder factory.
+ *
+ * @author Denis Stepanov
+ * @since 4.0.0
+ */
+public abstract class BeanDefinitionBuilderFactory {
```

Comment:
Don't think we need another class here, just move these static methods to `BeanDefinitionBuilder` (or better named `BeanDefinitionCreator`

Reply:
We cannot have non-public methods in the interface, and I don't want to expose those helper methods.